screens/changePassword/ChangePasswordScreen.tsx

```diff
@@ -1,68 +1,131 @@
+  const [newPassword, setNewPassword] = useState("")
+  const [reEnteredPassword, setReEnteredPassword] = useState("")
+
+  const passwordResult = Password.validate(newPassword)
+
+  const mutation = useMutation(
+    async () => {
+      if (passwordResult) {
+        return await onPasswordChangeSubmitted(currentPassword, passwordResult)
+      }
+    },
+    {
+      onError: () => {
+        Alert.alert(
+          "Whoops",
+          "Sorry, something went wrong when trying to change your password. Please try again.",
+          [
+            { text: "Try Again", onPress: () => mutation.mutate() },
+            { text: "Ok" }
+          ]
+        )
+      }
+    }
+  )
+
+  console.log(mutation.isLoading)
+
+  const errorReason = () => {
+    if (currentPassword === newPassword) {
+      return { status: "invalid", error: "current-matches-new" }
+    } else if (reEnteredPassword !== newPassword) {
+      return { status: "invalid", error: "reenter-does-not-match-new" }
+    } else if (!passwordResult) {
+      return { status: "invalid", error: "weak-new-password" }
+    } else if (mutation.isLoading) {
+      return { status: "submitting" }
+    } else if (mutation.data === "incorrect-password") {
+      return { status: "invalid", error: "incorrect-current-password" }
+    } else {
+      return { status: "valid", submit: mutation.mutate }
+    }
+  }
+
+  // console.log(errorReason())
+
+  return {
+    currentPassword,
+    setCurrentPassword,
+    newPassword,
+    setNewPassword,
+    reEnteredPassword,
+    setReEnteredPassword,
+    submission: errorReason()
+  }
+}
+
 type FormSubmission =
   | { status: "valid"; submit: () => void }
   | { status: "invalid"; errors: string[] }
 
-export const ChangePasswordScreen = ({
-  onPasswordChangeSubmitted = changePassword
-}: ChangePasswordProps) => {
-  const [currentPassword, setCurrentPassword] = useState<string>("")
-  const [newPassword, setNewPassword] = useState<string>("")
-  const [reEnteredPassword, setReEnteredPassword] = useState<string>("")
-
-  const [currentPasswordHidden, setCurrentPasswordHidden] =
-    useState<boolean>(true)
-  const [newPasswordHidden, setNewPasswordHidden] = useState<boolean>(true)
-  const [reEnteredPasswordHidden, setReEnteredPasswordHidden] =
-    useState<boolean>(true)
-
-  const validateForm = () => {
-    const formValid: boolean =
-      oldAccountPassword === currentPassword &&
-      validateNewPassword(newPassword) &&
-      reEnteredPassword === newPassword
-    isValidForm = formValid
-    return formValid
-  }
-
+export const ChangePasswordScreen = () => {
   // Function activated on button tap
-  const tapChangePassword = () => {
-    const submit = () => {
-      onPasswordChangeSubmitted(currentPassword, newPassword)
-    }
-    const submission: FormSubmission = isValidForm
-      ? { status: "valid", submit }
-      : { status: "invalid", errors: [] }
-    return { currentPassword, submission }
-  }
+  const {
```

Comment:
Have the hook call outside of the component, in order to keep the component headless